src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Stringable;
+
+class Url implements Arrayable, Stringable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public ?string $scheme = null,
+        public ?string $host = null,
+        public ?string $port = null,
+        public ?string $user = null,
+        public ?string $pass = null,
+        public ?string $path = null,
+        public ?string $query = null,
+        public ?string $fragment = null,
```

Comment:
`parse_url` returns port as an int. Is it worth casting it to a string here?

Maybe it's worth to reflect the types according to a spec, e.g. https://url.spec.whatwg.org/#url-representation

Reply:
I've updated the type to `int`. Thanks!